oak_runtime/src/node/http/mod.rs

```diff
@@ -301,19 +301,45 @@ struct HttpRequestHandler {
 impl HttpRequestHandler {
     async fn handle(&self, req: Request<Body>) -> Result<Response<Body>, OakStatus> {
         let request = HttpServerNode::map_to_http_request(req).await;
-        let oak_label = get_oak_label(&request)?;
-        info!(
-            "Handling HTTP request; request size: {} bytes, label: {:?}",
-            request.body.len(),
-            oak_label
-        );
-
-        debug!("Inject the request into the Oak Node");
-        let response = self
-            .inject_http_request(request, &oak_label)
-            .map_err(|_| OakStatus::ErrInternal)?;
-
-        Ok(response.to_response())
+        match get_oak_label(&request) {
+            Ok(oak_label) => {
+                info!(
+                    "Handling HTTP request; request body size: {} bytes, label: {:?}",
+                    request.body.len(),
+                    oak_label
+                );
+
+                debug!("Injecting the request into the Oak Node");
+                let response = self
+                    .inject_http_request(request, &oak_label)
+                    .map_err(|err| {
+                        warn!(
+                            "Error when injecting the request into the Oak Node: {:?}",
+                            err
+                        );
+                        OakStatus::ErrInternal
+                    })?;
+
+                response.to_response().map_err(|e| {
+                    warn!("Could not create response: {}", e);
+                    OakStatus::ErrInternal
+                })
+            }
+            Err(OakStatus::ErrInvalidArgs) => http::response::Builder::new()
+                .status(StatusCode::BAD_REQUEST)
+                .body(Body::from("Invalid or missing Oak label."))
+                .map_err(|e| {
+                    warn!("Could not create response: {}", e);
+                    OakStatus::ErrInternal
+                }),
+            Err(_oak_status) => http::response::Builder::new()
```

Comment:
@tiziano88 Do you think we need a separate match arm and status code for each `OakStatus`? I am not even sure what the mapping would be, since we are not even using all OakStatuses. Perhaps it is best to define a separate Error type here. WDYT?

Reply:
No, I think the current logic is fine. Basically we only distinguish between user error and internal error. An alternative may be to define a mapping from `OakStatus` to `StatusCode`, similar to what I think we already have for gRPC status codes.

Reply:
And return a `StatusCode` from `get_oak_label` instead of `OakStatus`?

Reply:
No, keep returning OakStatus everywhere as it currently is, but in this function convert from OakStatus to StatusCode.